### DISCUSSION TITLE
How do we launch an app with logged in state?

### DISCUSSION THREAD

**Discussion by vanessafsoares** [wix/Detox#4024] [Q&A]
I've been trying to use the `device.launchApp()` function to set the login state while detox launches the app, but I don't know if it's possible to set an `accessToken `that way, has anyone tried something similar and it worked? Or know how we can use detox to do this?

**Comment by jzaefferer** [comment]
Our solution looks like this:
```
await device.launchApp({
  url: `detox:${email}:${password}`,
});
```
The `url` is then handled as a special deep link to fill the login fields and submit it (via JavaScript inside a webview). It seems quite hacky, but it works.
